### ISSUE TITLE
Shared Memory?

### ISSUE THREAD

**Issue by McFarts** [audreyt/node-webworker-threads#66]
Or is it the same API as the IPC?  I really don't want to send data across processes all the time when the memory can just be shared and forked.

**Comment by audreyt** [comment]
The proposed API is here: https://lars-t-hansen.github.io/ecmascript_sharedmem/shmem.html

( Source: https://github.com/lars-t-hansen/ecmascript_sharedmem )

We would need to port to `nan 2.0.x` in order to use this feature.

Also, would you like to help checking if SharedArrayBuffer is adopted by Node.js 4.0.x? Is it available in JS land?

**Comment by McFarts** [comment]
I don't think it will be adopted in nodejs.. maybe JXcore though.  I am trying to get JXcore to adopt it into their core [here](https://github.com/jxcore/jxcore/issues/535#issuecomment-139675245).

I will keep an eye out for it though.  There is also
https://github.com/lpetre/node-shm-buffer

And
http://blog.varunajayasiri.com/shared-memory-with-nodejs

That both work absolutely fine on any v8.

Problem is, it only works under Linux, not Windows.  So it's bad for development (if you dev on windows, but this is beyond the scope I guess)

Thanks for the response ~

**Comment by jorrit** [comment]
node-shm-buffer is also not using nan 2, unfortunately.

**Comment by netgusto** [comment]
The README.md is a bit misleading on the matter, given the fact that one cannot share memory between workers using `webworker-threads`.

> 4.- The threads of a process share exactly the same address space, that of the process they belong to. Every thread can access every memory address within the process' address space. This is a very appropriate setup when the threads are actually part of the same job and are actively and closely cooperating with each other. Passing a reference to a chunk of data via a pointer is many orders of magnitude faster than transferring a copy of the data via IPC.

Granted, this explains threads and not `webworker-threads`, but nonetheless :)

**Comment by audreyt** [comment]
This is true. I've removed that paragraph until we do implement transferable / shared memory features.

**Comment by wagenaartje** [comment]
Any updates on this?
